### PR TITLE
tests.texlive: several new tests

### PR DESCRIPTION
**Pull request by xworld21**
## Description of changes
Implement a generic `tests.texlive.mkTeXTest` and use it to add a bunch of new tests related to hyphenation patters.

## Things done
- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xworld21** [comment]
@GrahamcOfBorg build tests.texlive

File: pkgs/test/texlive/default.nix [hunk 1]
```diff
@@ -1,6 +1,31 @@
 { lib, stdenv, buildEnv, runCommand, fetchurl, file, texlive, writeShellScript, writeText }:
 
-{
+rec {
+
+  mkTeXTest = lib.makeOverridable (
+    { name
+    , format
+    , tex
```
Comment (apfelkuchen6) [inline pkgs/test/texlive/default.nix:8]:
```suggestion
    , text
```
Maybe `text` (as in `pkgs.writeTextFile` ) is a better name for this argument as `tex` could  me misinterpreted more easliy.

**Review by apfelkuchen6 — APPROVED** [review]

**Comment by xworld21** [comment]
Just a warning, while renaming `tex` to `text`, I have changed the scheme of the opentype tests to `scheme-small`. Always check:

@GrahamcOfBorg build tests.texlive

@veprbl these new tests are quite handy, best if you merge the test-related patches (this one, and the fixed hashes test) before the other big things.

**Review by veprbl — APPROVED** [review]
LGTM

**Pull request merged**